### ISSUE TITLE
Arithmetic differences: add.bizdays vs bizdays

### ISSUE THREAD

**Issue by bbrewington** [wilsonfreitas/R-bizdays#63]
When you add 1 business day to 10/15/17 (a Sunday), it returns 10/16/17 (a Monday)

```{r}
create.calendar(name = "MyCalendar", weekdays = c("saturday", "sunday"))

add.bizdays(dates = as.Date("2017-10-15"), 1, cal = "MyCalendar")
# returns 2017-10-16

bizdays(from = as.Date("2017-10-15"), to = as.Date("2017-10-16"), cal = "MyCalendar")
# returns 0
```

However, when you find the business days between 10/15/17 & 10/16/17, it is 0.  Shouldn't `add.bizdays` and `bizdays` be inverses of each other?

**Comment by andreobereigner** [comment]
Hello Wilson,

the above issue may be related to the following (using your Vignette "Counting_business_days.Rmd" and the visual illustration in it):

> create.calendar(name = "example1", weekdays = c("saturday", "sunday"), start.date = "2017-01-24", end.date = "2017-01-30", holidays = "2017-01-25")
> calendars() # to list all calendars
Calendars: 
actual, Brazil/ANBIMA, example1, weekends

**> # Instead of 1, the function returns 0.**
> bizdays("2017-01-24", "2017-01-25", "example1")
[1] 0

**> # Instead of 2, the function returns 1.**
> bizdays("2017-01-24", "2017-01-26", "example1")
[1] 1

**Comment by wilsonfreitas** [comment]
Hi @bbrewington and @andreobereigner I am really thank for your points and sorry for taking so long to return, but I am back now and lets get this issue answered.

@andreobereigner 
The problem you pointed really used to happen and I've discovered this while the vignette you mentioned was being written and this problem was solved in bizdays 1.0.2 with the implementation of the double index algorithm to compute the business days. The code below must return 0 for the two calls.

```r
bizdays("2017-01-24", "2017-01-25", "example1")
# [1] 0
bizdays("2017-01-25", "2017-01-26", "example1")
# [1] 0
```
I would ask you to get bizdays updated and try again.

@bbrewington 

Your point is more difficult and I don't have a good answer.
I believe that a convention must be placed in order to avoid awkward issues.

The call   

```r
add.bizdays(dates = as.Date("2017-10-15"), 1, cal = "MyCalendar")
```

returns the next bizday according to the given calendar. While the other call

```r
bizdays(from = as.Date("2017-10-15"), to = as.Date("2017-10-16"), cal = "MyCalendar")
```

returns 0 because `from` and `to` dates share the same index and this makes much sense to me because once we are counting business days between dates I would say that between "2017-10-15" and "2017-10-16" we don't have one bizday.

All this happens when a non business day is used as argument of `bizdays` and `add.bizdays` functions. This is a weakness of bizdays counting algorithm and some conventions must be assumed. This is also the reason why I introduced the arguments `adjust.to` and `adjust.from`, I have plans to deprecate them, but I am not that sure.

If this problem is annoying you I suggest you call `bizdays` and `add.bizdays` using `adjust.previous` and `adjust.next` every time it is appropriate. For example

```r
adjust.previous("2017-10-15", "MyCalendar")
# [1] "2017-10-13"
add.bizdays(adjust.previous("2017-10-15"), 1, "MyCalendar")
# [1] "2017-10-16"
bizdays(from = adjust.previous("2017-10-15"), to = "2017-10-16", cal = "MyCalendar")
# [1] 0
```

**Comment by wilsonfreitas** [comment]
I have one possible solutions for this issue and I'd like to hear from you.

The add.bizdays function could return `NA` every time a non-business day is passed to add.bizdays and also raise a warning.
This might sound non sense try to add bizdays to a non business day and the user should be warned about that.

@bbrewington @andreobereigner 
What do you think?
